### PR TITLE
WIP: 5.0.0 breaking changes

### PR DESCRIPTION
**Pull request by jjasghar**
This PR is for bumping to 5.0.0 and the braking changes.

- Update kitchen-dokken.yml
- Added support for Debian 8
- Bump version of RabbitMQ
- Getting RHEL platform family working again
- RHEL and systemd have issues starting Rabbitmq put some logic around testing and starting it automatically by default.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -24,7 +25,11 @@ class Chef::Resource
   include Opscode::RabbitMQ # rubocop:enable all
 end
 
-include_recipe 'erlang'
+case node['platform_family']
+when 'rhel'
+else
+  include_recipe 'erlang'
```
Comment (tas50) [inline recipes/default.rb]: include_recipe 'erlang' unless platform_family?('rhel')

File: recipes/default.rb [hunk 2]
```diff
@@ -111,6 +114,28 @@ class Chef::Resource
       version node['rabbitmq']['version'] if node['rabbitmq']['pin_distro_version']
     end
   else
+    package 'wget' do
+      action :install
```
Comment (tas50) [inline recipes/default.rb]: you can skip the whole action part since :install is the default

File: recipes/default.rb [hunk 3]
```diff
@@ -111,6 +114,28 @@ class Chef::Resource
...
+        rpm -Uvh erlang-solutions-1.0-1.noarch.rpm      || STATUS=1
+      exit $STATUS
+      EOH
+    end
+
+    # socat is a package dependency of rabbitmq-server
+    package 'socat'
+
+    package 'erlang' do
+      action :install
```
Comment (tas50) [inline recipes/default.rb]: same thing with this one

File: recipes/default.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# coding: utf-8
```
Comment (jjasghar) [inline recipes/default.rb]: updated 🤘

File: recipes/default.rb (hunk 1 above)
Comment (jjasghar) [inline recipes/default.rb]: updated 🤘

File: recipes/default.rb (hunk 2 above)
Comment (jjasghar) [inline recipes/default.rb]: updated 🤘

File: recipes/default.rb (hunk 3 above)
Comment (jjasghar) [inline recipes/default.rb]: updated 🤘

File: recipes/default.rb [hunk 5]
```diff
@@ -87,16 +87,13 @@ class Chef::Resource
       source 'rabbitmq.upstart.conf.erb'
       owner 'root'
       group 'root'
-      mode 0644
+      mode 0o644
```
Comment (devsibwarra) [inline recipes/default.rb]: typo?

Comment (tas50) [inline recipes/default.rb]: Cookstyle autocorrect. The new version will not do this. Instead we should set this to '644'

**Pull request closed without merging**